src/webgl/uniforms.js

```diff
@@ -135,53 +118,37 @@ const UNIFORM_SETTERS = {
 };
 
 // Pre-allocated typed arrays for temporary conversion
-const FLOAT_ARRAY = [2, 3, 4, 6, 8, 9, 12, 16].reduce((arrays, length) => {
-  arrays[length] = new Float32Array(length);
-  return arrays;
-}, {});
-const INT_ARRAY = {
-  2: new Int32Array(2),
-  3: new Int32Array(3),
-  4: new Int32Array(4)
-};
-const UINT_ARRAY = {
-  2: new Uint32Array(2),
-  3: new Uint32Array(3),
-  4: new Uint32Array(4)
-};
+const FLOAT_ARRAY = {};
+const INT_ARRAY = {};
+const UINT_ARRAY = {};
 
 /* Functions to ensure the type of uniform values */
-function toFloatArray(value, length) {
```

Comment:
I'd keep the `length` parameter and verify that the provided array was an integral multiple of that length.

Reply:
What do we do if it's not? Error seems a little intolerant. Crop? Warn?

Reply:
Yes, I think an (automatically deduplicated) warning would be appropriate. I see it as one of the strengths of luma.gl that inputs to the GPU are validated in JS, this would fit with that spirit.